storage/remote/client.go

```diff
@@ -199,25 +205,77 @@ type RecoverableError struct {
 	retryAfter model.Duration
 }
 
+// Attempt a HEAD request against a remote write endpoint to see what it supports.
+func (c *Client) GetProtoVersions(ctx context.Context) (string, error) {
+	// If we are in Version1 mode then don't even bother
+	if c.rwFormat == Version1 {
+		return RemoteWriteVersion1HeaderValue, nil
+	}
+
+	httpReq, err := http.NewRequest("HEAD", c.urlString, nil)
+	if err != nil {
+		// Errors from NewRequest are from unparsable URLs, so are not
+		// recoverable.
+		return "", err
+	}
+
+	// Set the version header to be nice
+	httpReq.Header.Set(RemoteWriteVersionHeader, RemoteWriteVersion20HeaderValue)
+	httpReq.Header.Set("User-Agent", UserAgent)
+
+	ctx, cancel := context.WithTimeout(ctx, c.timeout)
+	defer cancel()
+
+	httpResp, err := c.Client.Do(httpReq.WithContext(ctx))
+	if err != nil {
+		// We don't attempt a retry here
+		return "", err
+	}
+
+	// See if we got a header anyway
```

Comment:
I don't want to be picky, but all those comments would be better if they have a trailing dot 🙈 
```suggestion
	// See if we got a header anyway.
```

Reply:
I think linting should actually fail here :thinking: